flac.go

```diff
@@ -253,7 +278,99 @@ func (stream *Stream) ParseNext() (f *frame.Frame, err error) {
 	return frame.Parse(stream.r)
 }
 
-// Seek is not implement yet.
-func (stream *Stream) Seek(offset int64, whence int) (read int64, err error) {
-	return 0, nil
+// Seek to offset where offset represents the sample number in the flac file
+//
+// TODO: return an error if sample is invalid?  Do nothing?
+//
+// sample could be invalid if:
+// whence == io.SeekEnd && sample is not negative
+// whence == io.SeekStart && sample is negative
+// whence == io.SeekCurrent and sample + current sample < 0 or > stream.Info.NSamples
+func (stream *Stream) Seek(sample int64, whence int) (read int64, err error) {
+	if stream.seekTable == nil {
+		return 0, nil
+	}
+
+	r := stream.r.(io.ReadSeeker)
+
+	var point meta.SeekPoint
+	switch whence {
+	case io.SeekStart:
+		point = stream.searchFromStart(sample)
+	case io.SeekCurrent:
+		o, err := r.Seek(0, io.SeekCurrent)
+		if err != nil {
+			return 0, err
+		}
+		point = stream.searchFromCurrent(sample, o)
+	case io.SeekEnd:
+		point = stream.searchFromEnd(sample)
+	default:
+		return 0, fmt.Errorf("invalid whence %d, must one of %d, %d or %d", whence, io.SeekStart, io.SeekCurrent, io.SeekEnd)
+	}
+
+	_, err = r.Seek(stream.start+int64(point.Offset), io.SeekStart)
+	return int64(point.SampleNum), err
+}
+
+func (stream *Stream) searchFromCurrent(sample, offset int64) (p meta.SeekPoint) {
```

Comment:
suggestion: ignore this comment if this function call is considered fast.

Reply:
I think it's still O(n) because worst case you are still looping through the seekTable 2x, right?  This isn't a double loop, even though it kind of looks like it.

Reply:
Ah, you're right! Then it's probably fine :D